addons/purchase_stock/models/purchase_order_line.py

```diff
@@ -273,11 +284,14 @@ def _check_orderpoint_picking_type(self):
             raise UserError(_('The warehouse of operation type (%(operation_type)s) is inconsistent with location (%(location)s) of reordering rule (%(reordering_rule)s) for product %(product)s. Change the operation type or cancel the request for quotation.',
                               product=self.product_id.display_name, operation_type=self.order_id.picking_type_id.display_name, location=self.orderpoint_id.location_id.display_name, reordering_rule=self.orderpoint_id.display_name))
 
-    def _prepare_stock_move_vals(self, picking, price_unit, product_uom_qty, product_uom):
+    def _prepare_stock_move_vals(self, price_unit, product_uom_qty, product_uom, picking=False):
         self.ensure_one()
         self._check_orderpoint_picking_type()
-        product = self.product_id.with_context(lang=self.order_id.dest_address_id.lang or self.env.user.lang)
-        location_dest = self.env['stock.location'].browse(self.order_id._get_destination_location())
+        if not self.order_id.reference_ids:
+            self.order_id.reference_ids = self.order_id.reference_ids.create(self.order_id._prepare_reference_vals())
+        if not self.order_id.partner_id.property_stock_supplier.id:
+            raise UserError(_("You must set a Vendor Location for this partner %s", self.partner_id.name))
```

Comment:
Better to name parameters to give more context to translators
```suggestion
            raise UserError(_("You must set a Vendor Location for partner %(partner_name)s", partner_name=self.partner_id.name))
```